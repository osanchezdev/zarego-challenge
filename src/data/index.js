export const ADMISION_TYPES = [
  'Freshman Student',
  'Transfer Student',
  'Graduate Student',
  'International Student',
];

export const COLLEGES_DATA = [
  'Bethesda University',
  'College of Alameda',
  'California College San Diego',
  'Columbia College',
  'Foothill College',
  'Golden West College',
  'Hartnell College',
  'ICOHS College',
  'Oxnard College',
  'Pitzer College',
];
