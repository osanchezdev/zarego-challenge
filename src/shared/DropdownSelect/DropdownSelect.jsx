import React from 'react';
import { bool, string, arrayOf, func } from 'prop-types';
import styles from './DropdownSelect.module.css';

const DropdownSelect = ({
  label,
  name,
  controller,
  showLabel,
  error,
  errorMsg,
  data,
}) => (
  <div className={styles.dropdown_select__wrapper}>
    {label && showLabel && (
      <div className={styles.select_label__wrapper}>
        <label htmlFor={name} className={styles.select_label}>
          {label}
        </label>
      </div>
    )}
    <div className={styles.select__wrapper}>
      <select
        id={name}
        defaultValue="null"
        name={name}
        className={`${styles.select} ${error ? styles.select_error : ''}`}
        {...controller(name)}
      >
        {data && data.length ? (
          <>
            <option disabled value="null">
              Select
            </option>
            {data.map((item) => (
              <option key={item} className={styles.select_option} value={item}>
                {item}
              </option>
            ))}
          </>
        ) : (
          <option value="null" readOnly className={styles.select_option_empty}>
            No options
          </option>
        )}
      </select>
    </div>
    {error && <p className={styles.select_error_msg}>{errorMsg}</p>}
  </div>
);

DropdownSelect.propTypes = {
  label: string.isRequired,
  name: string.isRequired,
  controller: func.isRequired,
  showLabel: bool,
  error: bool,
  errorMsg: string,
  data: arrayOf(string).isRequired,
};

DropdownSelect.defaultProps = {
  showLabel: true,
  error: false,
  errorMsg: null,
};

export default DropdownSelect;
