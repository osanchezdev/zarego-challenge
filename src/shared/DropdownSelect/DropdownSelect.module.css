.dropdown_select__wrapper {
  position: relative;
}

.select_label__wrapper {
  margin-bottom: 15px;
}

.select__wrapper {
  position: relative;
}

.select {
  height: 36px;
  width: 100%;
  min-width: 190px;
  font-size: 20px;
  color: #666;
}

.select.select_error {
  border: 2px inset #dc3545;
}

.select_option {
}

.select_error_msg {
  position: relative;
  padding: 0;
  flex: 0 0 100%;
  position: relative;
  margin: 0;
  color: #dc3545;
}
