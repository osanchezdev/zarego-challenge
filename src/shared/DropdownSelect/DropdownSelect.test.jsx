import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import DropdownSelect from './DropdownSelect';

import { ADMISION_TYPES as testData } from '../../data';

const controller = () => ({
  ref: jest.fn(),
  onBlur: jest.fn(),
  onChange: jest.fn(),
});
const { labelText, name } = {
  labelText: 'labelTest',
  name: 'nameField',
};

describe('DropdownSelect component snapshot', () => {
  it('should render component', () => {
    const tree = renderer
      .create(
        <DropdownSelect
          label={labelText}
          name={name}
          data={testData}
          controller={controller}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('DropdownSelect component funtional tests', () => {
  beforeAll(() =>
    render(
      <DropdownSelect
        label={labelText}
        name={name}
        data={testData}
        controller={controller}
      />
    )
  );

  it('should render all options', () => {
    const options = screen.queryAllByRole('option');
    expect(options).toHaveLength(testData.length + 1);
  });
});
