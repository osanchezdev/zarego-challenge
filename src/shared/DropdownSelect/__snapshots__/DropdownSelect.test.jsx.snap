// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownSelect component snapshot should render component 1`] = `
<div
  className="dropdown_select__wrapper"
>
  <div
    className="select_label__wrapper"
  >
    <label
      className="select_label"
      htmlFor="nameField"
    >
      labelTest
    </label>
  </div>
  <div
    className="select__wrapper"
  >
    <select
      className="select "
      defaultValue="null"
      id="nameField"
      name="nameField"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
    >
      <option
        disabled={true}
        value="null"
      >
        Select
      </option>
      <option
        className="select_option"
        value="Freshman Student"
      >
        Freshman Student
      </option>
      <option
        className="select_option"
        value="Transfer Student"
      >
        Transfer Student
      </option>
      <option
        className="select_option"
        value="Graduate Student"
      >
        Graduate Student
      </option>
      <option
        className="select_option"
        value="International Student"
      >
        International Student
      </option>
    </select>
  </div>
</div>
`;
